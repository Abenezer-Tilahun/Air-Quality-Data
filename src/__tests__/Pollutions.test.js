import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import countryReducer from '../Redux/Reducers/Country';
import pollutionReducer from '../Redux/Reducers/Pollution';
import Pollutions from '../components/Homepage/Pollutions';

const rootReducer = combineReducers({
  countryReducer,
  pollutionReducer,
});
const store = createStore(rootReducer, applyMiddleware(thunk));

afterEach(cleanup);

describe('Pollutions List ', () => {
  test('Pollution lists render correctly', () => {
    const pollutions = render(
      <Provider store={store}>
        <Router>
          <Pollutions />
        </Router>
      </Provider>,
    );
    expect(pollutions).toMatchSnapshot();
  });
});
