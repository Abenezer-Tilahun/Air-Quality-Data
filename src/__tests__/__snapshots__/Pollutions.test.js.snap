// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pollutions List  Pollution lists render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="bg-pink-500 shadow-lg"
        >
          <nav>
            <div
              class="d-flex justify-content-between align-items-center p-3 mb-2"
            >
              <button
                class="btn mt-2"
                type="button"
              >
                <a
                  aria-current="page"
                  class="link m-2 text-light active"
                  href="/"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-house icon"
                    data-icon="house"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
                      fill="currentColor"
                    />
                  </svg>
                </a>
              </button>
              <h1
                class="header"
              >
                AirNow
              </h1>
              <a
                class="link text-light"
                href="/countries"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle-arrow-left icon"
                  data-icon="circle-arrow-left"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  text="reload"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM384 288H205.3l49.38 49.38c12.5 12.5 12.5 32.75 0 45.25s-32.75 12.5-45.25 0L105.4 278.6C97.4 270.7 96 260.9 96 256c0-4.883 1.391-14.66 9.398-22.65l103.1-103.1c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L205.3 224H384c17.69 0 32 14.33 32 32S401.7 288 384 288z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </div>
          </nav>
        </div>
        <div
          class="pollution m-5"
        >
          <h3
            class="text-white m-2"
          />
          <div
            class="m-2"
          >
            <img
              alt="flag"
              class="w-25"
            />
          </div>
          <p
            class="m-2 text-warning"
          >
            Air pollution ( μg/m3 ) stats - 2022:
          </p>
          <table
            class="table"
          >
            <thead>
              <tr
                class="table-light"
              >
                <th
                  scope="col"
                >
                  gas
                </th>
                <th
                  scope="col"
                >
                  μg/m3
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="table-secondary"
              >
                <th
                  scope="row"
                >
                  CO
                </th>
                <td />
              </tr>
              <tr
                class="table-danger"
              >
                <th
                  scope="row"
                >
                  NO
                </th>
                <td />
              </tr>
              <tr
                class="table-warning"
              >
                <th
                  scope="row"
                >
                  NO2
                </th>
                <td />
              </tr>
              <tr
                class="table-info"
              >
                <th
                  scope="row"
                >
                  NO2
                </th>
                <td />
              </tr>
              <tr
                class="table-info"
              >
                <th
                  scope="row"
                />
                <td />
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="bg-pink-500 shadow-lg"
      >
        <nav>
          <div
            class="d-flex justify-content-between align-items-center p-3 mb-2"
          >
            <button
              class="btn mt-2"
              type="button"
            >
              <a
                aria-current="page"
                class="link m-2 text-light active"
                href="/"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-house icon"
                  data-icon="house"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </button>
            <h1
              class="header"
            >
              AirNow
            </h1>
            <a
              class="link text-light"
              href="/countries"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-arrow-left icon"
                data-icon="circle-arrow-left"
                data-prefix="fas"
                focusable="false"
                role="img"
                text="reload"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM384 288H205.3l49.38 49.38c12.5 12.5 12.5 32.75 0 45.25s-32.75 12.5-45.25 0L105.4 278.6C97.4 270.7 96 260.9 96 256c0-4.883 1.391-14.66 9.398-22.65l103.1-103.1c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L205.3 224H384c17.69 0 32 14.33 32 32S401.7 288 384 288z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </nav>
      </div>
      <div
        class="pollution m-5"
      >
        <h3
          class="text-white m-2"
        />
        <div
          class="m-2"
        >
          <img
            alt="flag"
            class="w-25"
          />
        </div>
        <p
          class="m-2 text-warning"
        >
          Air pollution ( μg/m3 ) stats - 2022:
        </p>
        <table
          class="table"
        >
          <thead>
            <tr
              class="table-light"
            >
              <th
                scope="col"
              >
                gas
              </th>
              <th
                scope="col"
              >
                μg/m3
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="table-secondary"
            >
              <th
                scope="row"
              >
                CO
              </th>
              <td />
            </tr>
            <tr
              class="table-danger"
            >
              <th
                scope="row"
              >
                NO
              </th>
              <td />
            </tr>
            <tr
              class="table-warning"
            >
              <th
                scope="row"
              >
                NO2
              </th>
              <td />
            </tr>
            <tr
              class="table-info"
            >
              <th
                scope="row"
              >
                NO2
              </th>
              <td />
            </tr>
            <tr
              class="table-info"
            >
              <th
                scope="row"
              />
              <td />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
